server/src/main/java/org/elasticsearch/common/unit/RelativeByteSizeValue.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.unit;
+
+import org.elasticsearch.ElasticsearchParseException;
+
+/**
+ * A byte size value that allows specification using either of:
+ * 1. Absolute value (200GB for instance)
+ * 2. Relative percentage value (95%)
+ * 3. Relative ratio value (0.95)
+ */
+public class RelativeByteSizeValue {
+
+    public static final String MAX_HEADROOM_PREFIX = "max_headroom=";
+    private final ByteSizeValue absolute;
+    private final RatioValue ratio;
+
+    public RelativeByteSizeValue(ByteSizeValue absolute) {
+        this.absolute = absolute;
+        this.ratio = null;
+    }
+
+    public RelativeByteSizeValue(RatioValue ratio) {
+        this.absolute = null;
+        this.ratio = ratio;
+    }
+
+    public boolean isAbsolute() {
+        return absolute != null;
+    }
+
+    public ByteSizeValue getAbsolute() {
+        return absolute;
+    }
+
+    public RatioValue getRatio() {
```

Comment:
This entire class and associated test is also in #71844, only difference is that this method was made public here.